File: CS/WebApplication2/Helpers/CustomEvents.cs

using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using Google.Apis.Calendar.v3;
using Google.Apis.Calendar.v3.Data;
using DevExpress.Web.ASPxScheduler;
using DevExpress.XtraScheduler;

[Serializable]
public class CustomEvent {
    Event eventEntry;
    string recurrenceInfo;
    public CustomEvent(Event eventEntry) {
        this.eventEntry = eventEntry;
    }
    public CustomEvent() {
        this.eventEntry = new Event();
    }

    public Event SourceEvent { get { return eventEntry; } }
    public string Subject { get { return SourceEvent.Summary; } set { SourceEvent.Summary = value; } }
    public DateTime StartTime {
        get { return CalendarApiHelper.ConvertDateTime(SourceEvent.Start); }
        set {
            if (eventEntry.Start == null)
                eventEntry.Start = new EventDateTime();
            eventEntry.Start.DateTime = value;
        }
    }
    public DateTime EndTime {
        get { return CalendarApiHelper.ConvertDateTime(SourceEvent.End); }
        set {
            if (eventEntry.End == null)
                eventEntry.End = new EventDateTime();
            eventEntry.End.DateTime = value;
        }
    }
    public string Description { get { return SourceEvent.Description; } set { eventEntry.Description = value; } }
    public string Location { get { return SourceEvent.Location; } set { eventEntry.Location = value; } }
    public bool AllDay {
        get { return CalendarApiHelper.IsAllDay(SourceEvent.Start); }
        set {
            if (AllDay == value)
                return;
            if (value) {
                CalendarApiHelper.MakeAllDay(SourceEvent.Start);
                CalendarApiHelper.MakeAllDay(SourceEvent.End);
            } else {
                CalendarApiHelper.MakeDate(SourceEvent.Start);
                CalendarApiHelper.MakeDate(SourceEvent.End);
                SourceEvent.End.DateTime = SourceEvent.End.DateTime.Value.Date.AddDays(1);
            }
        }
    }
    public object Id { get { return SourceEvent.Id; } set { eventEntry.Id = (value == null)? null : value.ToString(); } }
    public int Status { get; set; }
    public long Label { get; set; }
    public int EventType { get; set; }
  
}

public class CustomEventList : BindingList<CustomEvent> {
}

#region CustomEventDataSource
public class CustomEventDataSource {
    GoogleHelper googleHelper = null;
    public CustomEventDataSource(CalendarService calendarService, string calendarId, string timeZoneId)
    {
        this.googleHelper = new GoogleHelper(calendarService, calendarId, timeZoneId);
    }
    public CustomEventDataSource() {
        this.googleHelper = new GoogleHelper(null, String.Empty, string.Empty);
    }
    GoogleHelper GoogleHelper { get { return googleHelper; } }

    #region ObjectDataSource methods
    public string InsertMethodHandler(CustomEvent customEvent) {
        return GoogleHelper.InsertCustomEvent(customEvent);
    }
    public void DeleteMethodHandler(CustomEvent customEvent) {
        GoogleHelper.DeleteCustomEvent(customEvent);
    }
    public string UpdateMethodHandler(CustomEvent customEvent) {
        return GoogleHelper.UpdateCustomEvent(customEvent);
    }
    public IEnumerable SelectMethodHandler(DateTime startDate, DateTime endDate)
    {
        return GoogleHelper.SelectCustomEvent(startDate, endDate);
    }
   
    #endregion
}
#endregion

File: VB/WebApplication2/CustomEvents.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports Google.Apis.Calendar.v3
Imports Google.Apis.Calendar.v3.Data
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.XtraScheduler

<Serializable> _
Public Class CustomEvent
    Private eventEntry As [Event]
    Private recurrenceInfo As String
    Public Sub New(ByVal eventEntry As [Event])
        Me.eventEntry = eventEntry
    End Sub
    Public Sub New()
        Me.eventEntry = New [Event]()
    End Sub

    Public ReadOnly Property SourceEvent() As [Event]
        Get
            Return eventEntry
        End Get
    End Property
    Public Property Subject() As String
        Get
            Return SourceEvent.Summary
        End Get
        Set(ByVal value As String)
            SourceEvent.Summary = value
        End Set
    End Property
    Public Property StartTime() As Date
        Get
            Return CalendarApiHelper.ConvertDateTime(SourceEvent.Start)
        End Get
        Set(ByVal value As Date)
            If eventEntry.Start Is Nothing Then
                eventEntry.Start = New EventDateTime()
            End If
            eventEntry.Start.DateTime = value
        End Set
    End Property
    Public Property EndTime() As Date
        Get
            Return CalendarApiHelper.ConvertDateTime(SourceEvent.End)
        End Get
        Set(ByVal value As Date)
            If eventEntry.End Is Nothing Then
                eventEntry.End = New EventDateTime()
            End If
            eventEntry.End.DateTime = value
        End Set
    End Property
    Public Property Description() As String
        Get
            Return SourceEvent.Description
        End Get
        Set(ByVal value As String)
            eventEntry.Description = value
        End Set
    End Property
    Public Property Location() As String
        Get
            Return SourceEvent.Location
        End Get
        Set(ByVal value As String)
            eventEntry.Location = value
        End Set
    End Property
    Public Property AllDay() As Boolean
        Get
            Return CalendarApiHelper.IsAllDay(SourceEvent.Start)
        End Get
        Set(ByVal value As Boolean)
            If AllDay = value Then
                Return
            End If
            If value Then
                CalendarApiHelper.MakeAllDay(SourceEvent.Start)
                CalendarApiHelper.MakeAllDay(SourceEvent.End)
            Else
                CalendarApiHelper.MakeDate(SourceEvent.Start)
                CalendarApiHelper.MakeDate(SourceEvent.End)
                SourceEvent.End.DateTime = SourceEvent.End.DateTime.Value.Date.AddDays(1)
            End If
        End Set
    End Property
    Public Property Id() As Object
        Get
            Return SourceEvent.Id
        End Get
        Set(ByVal value As Object)
            eventEntry.Id = If(value Is Nothing, Nothing, value.ToString())
        End Set
    End Property
    Public Property Status() As Integer
    Public Property Label() As Long
    Public Property EventType() As Integer

End Class

Public Class CustomEventList
    Inherits BindingList(Of CustomEvent)

End Class

#Region "CustomEventDataSource"
Public Class CustomEventDataSource

    Private googleHelper_Renamed As GoogleHelper = Nothing
    Public Sub New(ByVal calendarService As CalendarService, ByVal calendarId As String, ByVal timeZoneId As String)
        Me.googleHelper_Renamed = New GoogleHelper(calendarService, calendarId, timeZoneId)
    End Sub
    Public Sub New()
        Me.googleHelper_Renamed = New GoogleHelper(Nothing, String.Empty, String.Empty)
    End Sub
    Private ReadOnly Property GoogleHelper() As GoogleHelper
        Get
            Return googleHelper_Renamed
        End Get
    End Property

    #Region "ObjectDataSource methods"
    Public Function InsertMethodHandler(ByVal customEvent As CustomEvent) As String
        Return GoogleHelper.InsertCustomEvent(customEvent)
    End Function
    Public Sub DeleteMethodHandler(ByVal customEvent As CustomEvent)
        GoogleHelper.DeleteCustomEvent(customEvent)
    End Sub
    Public Function UpdateMethodHandler(ByVal customEvent As CustomEvent) As String
        Return GoogleHelper.UpdateCustomEvent(customEvent)
    End Function
    Public Function SelectMethodHandler(ByVal startDate As Date, ByVal endDate As Date) As IEnumerable
        Return GoogleHelper.SelectCustomEvent(startDate, endDate)
    End Function

    #End Region
End Class
#End Region